m3/train/m3-vila8b.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# Set master address and worker list
+master_addr=$(hostname)
+export MASTER_ADDR=${master_addr:-"127.0.0.1"}
+export worker_list=$(hostname)
+
+echo "MASTER_ADDR="$MASTER_ADDR
+echo "JobID: $SLURM_JOB_ID | Full list: $worker_list"
+
+VILA_CODE_PATH="<path_to_vila_code>"
+
+# User-defined variables
+NODES=4
+GPUS_PER_NODE=8
+MASTER_PORT=25001
+STAGE2_PATH="/path/to/your/model"
+OUTPUT_DIR="/path/to/output/checkpoints"
+CONTAINER_IMAGE="/path/to/your/container/image.sqsh"
+CONTAINER_MOUNTS="/path/to/mounts"
+WANDB_API_KEY="your_wandb_api_key"
+
+# Removed srun and directly execute the commands
+source /root/miniconda3/bin/activate
+source /root/.bashrc
+cd $VILA_CODE_PATH
+conda activate vila
+echo `which python`
+wandb login <wandb_api_key>
+
+export PYTHONPATH=$VILA_CODE_PATH
+echo "PYTHONPATH is $PYTHONPATH"
+
+echo "MASTER_ADDR="$MASTER_ADDR
+CURRENT_RANK=${SLURM_PROCID:-"0"}
+n_node=${SLURM_JOB_NUM_NODES:-1}
+echo "JobID: $SLURM_JOB_ID | Full list: $worker_list | rank $CURRENT_RANK of $n_node"
+
+bs=16
+STAGE2_PATH=$STAGE2_PATH
+OUTPUT=provide_output_dir_name
+
+HEALTHCARE_DS=$(for i in {1..10}; do echo -n usmle+; done)
```

Comment:
Can you add a comment here in the scripts to describe what this is doing, e.g. "# UPSAMPLING FOR DATA BALANCING"